subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -56,15 +58,24 @@ struct fp_key_gen_account_key_check_context {
 };
 
 static uint8_t key_gen_failure_cnt;
-static struct k_work_delayable key_gen_failure_cnt_reset;
+static void key_gen_failure_cnt_reset_fn(struct k_work *w);
+K_WORK_DELAYABLE_DEFINE(key_gen_failure_cnt_reset, key_gen_failure_cnt_reset_fn);
 
-static bool user_pairing_mode = true;
+static bool user_pairing_mode;
 static struct fp_procedure fp_procedures[CONFIG_BT_MAX_CONN];
 
+static bool is_enabled;
 
-void bt_fast_pair_set_pairing_mode(bool pairing_mode)
+
+int bt_fast_pair_set_pairing_mode(bool pairing_mode)
```

Comment:
I wonder if we shouldn't let user control the pairing mode even if Fast Pair is disabled (in similar way as we do for battery level now)

Reply:
Ok, maybe it will allow more freedom for user.

Reply:
There seems to be no reason to disallow it